Standard Gaming Platform/vsurface.cpp

```diff
@@ -1801,42 +1790,24 @@ BOOLEAN AddVSurfaceRegions( HVSURFACE hVSurface, VSURFACE_REGION **ppNewRegions,
 	return( TRUE );
 }
 
-BOOLEAN RemoveVSurfaceRegion( HVSURFACE hVSurface, UINT16 usIndex )
-{
-	VSURFACE_REGION	aRegion;
-
-	Assert( hVSurface != NULL );
-
-	return( RemfromList( hVSurface->RegionList, &aRegion, usIndex ) );
-
-}
-
 BOOLEAN ClearAllVSurfaceRegions( HVSURFACE hVSurface )
 {
-	UINT32 uiListSize;
-
 	Assert( hVSurface != NULL );
-
-	uiListSize = ListSize( hVSurface->RegionList );
-
-	for ( INT32 cnt = uiListSize - 1; cnt >= 0; cnt-- )
-	{
-		RemoveVSurfaceRegion( hVSurface, (UINT16)cnt );
-	}
-
+	hVSurface->RegionList.clear();
 	return( TRUE );
 }
 
 BOOLEAN GetVSurfaceRegion( HVSURFACE hVSurface, UINT16 usIndex, VSURFACE_REGION *aRegion )
 {
 	Assert( hVSurface != NULL );
 
-	if ( !PeekList( hVSurface->RegionList, aRegion, usIndex ) )
+	if (usIndex < hVSurface->RegionList.size()-1)
 	{
-		return( FALSE );
+		*aRegion = hVSurface->RegionList[usIndex];
+		return( TRUE );
 	}
 
-	return( TRUE );
+	return( FALSE );

```

Comment:
Prefer to indent the happy path to the left. Namely, return false if usIndex is out of bounds and leave the `return true;` to be the last statement.